contracts/TownHallZap.sol

```diff
@@ -28,4 +36,61 @@ contract TownHallZap {
             townHall.mint(to);
         }
     }
+
+    // exactInputSingle for ERC20
+    function convertAndMint(address sourceToken, uint256 amountIn, uint24 _fee, address mintTo) external {
+        require(sourceToken != address(0), "sourceToken not address(0)");
+        require(mintTo.code.length == 0, "mintTo is not user");
+        require(IERC20(sourceToken).balanceOf(msg.sender) >= amountIn, "not enough sourceToken balance");
+
+        TransferHelper.safeTransferFrom(sourceToken, msg.sender, address(this), amountIn);
+        TransferHelper.safeApprove(sourceToken, address(uniswapV3Router), amountIn);
+
+        ISwapRouter.ExactInputSingleParams memory params = ISwapRouter.ExactInputSingleParams({
+            tokenIn: sourceToken,
+            tokenOut: address(huntToken),
+            fee: _fee,
```

Comment:
Can we just fix the `fee` parameter to `3000` ? because most HUNT LP pairs are on 0.3% fee

Reply:
fixed.